### PR TITLE
refactor: migrate collective long description to html

### PR DESCRIPTION
**Pull request by snitin315**
Refers https://github.com/opencollective/opencollective/issues/3207

Related Frontend PR https://github.com/opencollective/opencollective-frontend/pull/4357

Created a migration that takes all collectives with a markdown description and converts to HTML

### CODE REVIEW

File: migrations/20200606105847-convert-collective-long-description-to-html.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        WHERE c.id = :id
+      `,
+        {
+          replacements: {
+            longDescription: converter.makeMarkdown(collective.longDescription),
+            id: collective.id,
+          },
+        },
+      ),
+    );
```
Comment (Betree) [inline migrations/20200606105847-convert-collective-long-description-to-html.js:58]: The purpose of `down` is to rollback the changes if something wrong happens during the migration. What that would mean here is that we would need something to convert from HTML to markdown. We don't have that, and we probably don't want to introduce more complexity just to solve it. So doing this is fine:

```suggestion
    // Can't rollback this one
```

File: migrations/20200606105847-convert-collective-long-description-to-html.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import Promise from 'bluebird';
+import showdown from 'showdown';
+
+const converter = new showdown.Converter();
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    const collectives = await queryInterface.sequelize.query(
+      `
+        SELECT * FROM "Collectives"
```
Comment (Betree) [inline migrations/20200606105847-convert-collective-long-description-to-html.js:12]: In production, this query matches 6000+ collectives. Selecting the fields that we want will save some precious RAM:

```suggestion
        SELECT "id", "longDescription" FROM "Collectives"
```

File: migrations/20200606105847-convert-collective-long-description-to-html.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        WHERE LENGTH("longDescription") > 0 AND "longDescription" NOT LIKE '<%';
+      `,
```
Comment (Betree) [inline migrations/20200606105847-convert-collective-long-description-to-html.js:14]: No need to include deleted collectives

```suggestion
        WHERE LENGTH("longDescription") > 0 AND "longDescription" NOT LIKE '<%'
        AND "deletedAt" IS NULL;
      `,
```

File: migrations/20200606105847-convert-collective-long-description-to-html.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        WHERE c.id = :id
+      `,
+        {
+          replacements: {
+            longDescription: converter.makeHtml(collective.longDescription),
+            id: collective.id,
+          },
+        },
+      ),
+    );
```
Comment (Betree) [inline migrations/20200606105847-convert-collective-long-description-to-html.js:32]: As per the comment above, we have ~6000 collectives concerned by this migration. So this code may start a huge number of parallel DB queries. Could we rather have a `for` loop with an `await` inside (or similar) to make sure we do one query at a time?

Also if you have some insights on the time it takes to run locally, that'd be interesting for us to have.

**Comment by snitin315** [comment]
@Betree Updated :100:, Please take a look.

**Comment by Betree** [comment]
The migration runs ok on staging (~15s). I added a sanitize pass to make sure the result we generate matches what we have with the other editor.

**Pull request merged**